addons/io/org.openhab.io.transport.modbus/src/main/java/org/openhab/io/transport/modbus/ModbusBitUtilities.java

```diff
@@ -148,6 +149,48 @@ public static DecimalType extractStateFromRegisters(ModbusRegisterArray register
         }
     }
 
+    /**
+     * Read data from registers and convert the result to StringType
```

Comment:
Do you have opinion or experience with NUL characters in modbus data? Do we often have varying length string in Modbus? If it is the (common) case, perhaps this could stop reading the string when NUL is encountered or length has been reached? 

I'm also fine just documenting the handling of NUL or the fact that length should be defined such that any possible NUL character terminating the string is not included

Reply:
As far as I can tell using strings is not common with modbus anyways. But if one uses surely it will be vendor dependent.
The first where I see this is the sunspec specification. Digging deep in the specification they say "NULL terminated or padded".
Do you think this should get a parameter as well?

Reply:
I've added null termination handling I believe this could be a common use case. If you'd like I'll add an option parameter for this.

Reply:
I think this is really good! Let's go without parameter -- it can be always added later if needed.

Can you please document the NUL handling in the javadoc?